provider/src/test/kotlin/org/gradle/kotlin/dsl/provider/LinePreservingSubstringTest.kt

```diff
@@ -47,4 +47,36 @@ class LinePreservingSubstringTest {
                 }""".replaceIndent())
         )
     }
+
+    @Test
+    fun `given ranges linePreservingBlankRange should blank lines`() {
+        val original = """
+            |// line 1
+            |// line 2
+            |buildscript {
+            |    // line 4
+            |}
+            |// line 6
+            |plugins {
+            |    // line 8
+            |}
+            |// line 10
+        """.trimMargin()
+        val buildscriptRange = original.indexOf("buildscript")..original.indexOf("}")
+        val pluginsRange = original.indexOf("plugins")..original.lastIndexOf("}")
+        assertThat(
+            original.linePreservingBlankRanges(listOf(buildscriptRange, pluginsRange)),
+            equalTo("""
+                |// line 1
+                |// line 2
+                |
+                |
+                |
+                |// line 6
+                |
+                |
+                |
+                |// line 10
+            """.trimMargin()))
+    }
```

Comment:
Can you explain what this is supposed to be used for? I don't understand from the rest of the PR. It's cool, but I don't get why this is important.

Reply:
This is so line numbers are correctly reported in compilation error messages and exception stack traces.